src/components/Images/Images.js

```diff
@@ -1,31 +1,81 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import * as Cloudinary from 'cloudinary-react'
+import cloudinary from 'cloudinary-core'
+import useRequired from '../../hooks/useRequired.js'
+import useInvalid from '../../hooks/useInvalid.js'
 
 // https://cloudinary.com/documentation/image_transformation_reference
 
 export const CLOUDINARY_CLOUD_NAME = 'getethos'
 
-const publicIdBase = `https://res.cloudinary.com/${CLOUDINARY_CLOUD_NAME}/image/upload/`
+const cld = new cloudinary.Cloudinary({
+  cloud_name: CLOUDINARY_CLOUD_NAME,
+})
+
+const BREAKPOINTS = [600, 900, 1200]
+
+export const CloudinaryImage = ({
+  publicId,
+  className,
+  alt,
+  width,
+  height,
+  ...rest
+}) => {
+  // Verify that all required props were supplied
+  const [includesRequired] = useRequired(['publicId', 'alt'])
+  let allRelevantProps = Object.assign({}, rest, {
+    publicId,
+    alt,
+  })
+  includesRequired(allRelevantProps)
+
+  // Verify that no invalid props were supplied
+  const [includesInvalid] = useInvalid(
+    Object.keys(CloudinaryImage.PUBLIC_PROPS)
+  )
+  includesInvalid(rest)
+
+  const srcSetOptions = Object.assign(
+    {},
+    {
+      // Find information on image transformations here:
+      // https://cloudinary.com/documentation/image_transformations
+      quality: 'auto',
+      fetchFormat: 'auto',
+      srcset: {
+        breakpoints: [600, 900, 1200],
+        sizes: true,
+      },
+      dpr: 'auto',
+      crop: 'fill', // ?
+      secure: 'true', // ?
+    },
+    !!width ? { width } : {},
```

Comment:
Can we use `Boolean()` here?

Reply:
now i'm curious, looking up which is faster. This will get called twice for every image so we should do whatever's faster

Reply:
Looks like `!!` is faster. http://jsben.ch/JWVSk http://jsben.ch/nko0g The difference is pretty small, so if you prefer `Boolean()` we can change it.